src/npc.cpp

```diff
@@ -394,6 +417,30 @@ bool Npc::getNextStep(Direction& dir, uint32_t& flags)
 	return getRandomStep(dir);
 }
 
+void Npc::setIdle(bool idle)
+{
+	if (isRemoved() || getHealth() <= 0) {
+		return;
+	}
+
+	isIdle = idle;
+
+	if (!isIdle) {
+		g_game.addCreatureCheck(this);
+	} else {
+		onIdleStatus();
+		Game::removeCreatureCheck(this);
+	}
+}
+
+void Npc::updateIdleStatus()
+{
+	bool status = spectators.size() == 0;
```

Comment:
IMO checking size is better, if you read it you will expect something like "did it clean the list or not" since it's a boolean we are taking.